### DISCUSSION TITLE
New idea: G34 command with ender 3 s1

### DISCUSSION THREAD

**Discussion by superior-julz** [mriscoc/Ender3V2S1#1289] [Ideas]
I have had my Ender 3 s1 for 3 years and have had professional firmware for around 6 months. 
I recently was looking into the G34 command to auto-level the gantry. I know this means taking off the timing/sync belt, but if that is what is required, I am okay with that. I was simply suggesting implementing this into professional firmware. 
P.S. I have heard that you need a motherboard with 5 stepper motor ports (1 X, 1 Y, 2 Z, 1 E) and was wondering if anyone has any motherboard that does this. A cool bonus would be for that motherboard to be powerful enough to run Klipper.

**Comment by TheWebMachine** [comment]
> A cool bonus would be for that motherboard to be powerful enough to run Klipper.

Every implementation I've seen where Klipper is "built-in" to the printer, the printer's primary control MCU and firmware is separate from the MCU/APU running Klipper. You wouldn't want the totality of the Klipper side of the software stack causing hiccups or delays in executing gcode on the steppers. Controlling the steppers during a print is a real-time affair. 

Yes, some companies put both MCUs on one custom-built mainboard, but the printer firmware and Klipper run separately from one another on their own discreet hardware on that mainboard and talk to one another over a normal serial link, just like you'd run Klipper or Octoprint external to the printer. I'm not aware of any Klipper printers where the printer's core controller firmware and Klipper are running simultaneously on the same MCU.

**Reply by superior-julz** [reply]
okay thank you

**Comment by TheWebMachine** [comment]
That said, I'm totally in favor of the possibility of G34. I'll have to check the board to see if my Z steppers are being independently controller or tied to a single stepper controller. I have the E3 S1 Plus; basically same as yours, just bigger build volume. So, it might be a worthwhile option. 

I know I've had to manually re-level my gantry at least once now (I think the head crashing into a part caused a skip of the sync belt). It was a pain to manually skip sync belt teeth while turning screws. 🤣 G34 would be awesome. Especially since the Plus' larger bed is unwieldy to keep level across the X axis because of V-channel roller wear slowly adding slop and throwing it off over time. I'm constantly re-tramming my bed because if it.

**Comment by AndySmithers** [comment]
But...G34 doesn't necessarily need individual z steppers. So long as the single Z-stepper is in UART mode, you can use the Mechanical Gantry levelling feature of Marlin.
I'm using this now on an Ender3 V2 with a BKR Mini E3 V3 board.
It would be good to have the G34 in the LCD menu though...
